common/remote/sockfile.cpp

```diff
@@ -5745,13 +5745,29 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
         }
 
-        if (_useSSL)
+        if (_connectMethod == SSLOnly || _connectMethod == SSLFirst || _connectMethod == UnsecureFirst)
         {
             if (sslep.port == 0)
-                throw createDafsException(DAFSERR_connection_failed, "Secure dafilesrv port not specified");
+                throw createDafsException(DAFSERR_serverinit_failed, "Secure dafilesrv port not specified");
 
-            if ( (_useSSL != UnsecureFirst) && (!securitySettings.certificate || !securitySettings.privateKey) )
-                throw createDafsException(DAFSERR_connection_failed, "SSL Certificate and/or Key file information not found in environment.conf");
+            if (_connectMethod == UnsecureFirst)
```

Comment:
I understand the logic now, if there's an SSL port, even if no certificate and can't fully take SSL connections, it can still socket accept and failover quickly this way.

So I think it's good as is, with warning.